### ISSUE TITLE
Jackson creates an extra element if field is annotated with `@XmlValue` and has a (unannotated) getter

### ISSUE THREAD

**Issue by ewirch** [FasterXML/jackson-dataformat-xml#418]
Test case:
```
import static org.junit.Assert.assertEquals;

import com.fasterxml.jackson.dataformat.xml.JacksonXmlModule;
import com.fasterxml.jackson.dataformat.xml.XmlMapper;
import com.fasterxml.jackson.dataformat.xml.annotation.JacksonXmlProperty;
import com.fasterxml.jackson.dataformat.xml.annotation.JacksonXmlRootElement;
import com.fasterxml.jackson.dataformat.xml.annotation.JacksonXmlText;
import com.fasterxml.jackson.module.jaxb.JaxbAnnotationModule;
import org.junit.Test;

import java.io.IOException;
import javax.xml.bind.annotation.XmlAttribute;
import javax.xml.bind.annotation.XmlRootElement;
import javax.xml.bind.annotation.XmlValue;

public class JacksonElementValueMappingTest {

    private static final String EXPECTED = "<ROOT><CHILD attr=\"attr_value\">text</CHILD></ROOT>";

    @XmlRootElement(name = "ROOT")
    public static class RootWithJaxbAnnotations {
        @XmlAttribute(name = "CHILD")
        JaxbChild child = new JaxbChild();

        public JaxbChild getChild() {
            return this.child;
        }

        public void setChild(JaxbChild child) {
            this.child = child;
        }
    }

    public static class JaxbChild {
        @XmlAttribute
        String attr = "attr_value";

        @XmlValue
        String el = "text";

        public String getAttr() {
            return this.attr;
        }

        public String getEl() {
            return this.el;
        }

        public void setAttr(String attr) {
            this.attr = attr;
        }

        public void setEl(String el) {
            this.el = el;
        }
    }

    @JacksonXmlRootElement(localName = "ROOT")
    public static class RootWithJacksonAnnotations {
        @JacksonXmlProperty(localName = "CHILD")
        JacksonChild child = new JacksonChild();

        public JacksonChild getChild() {
            return this.child;
        }

        public void setChild(JacksonChild child) {
            this.child = child;
        }
    }

    public static class JacksonChild {
        @JacksonXmlProperty(isAttribute = true)
        String attr = "attr_value";

        @JacksonXmlText
        String el = "text";

        public String getAttr() {
            return this.attr;
        }

        public String getEl() {
            return this.el;
        }

        public void setAttr(String attr) {
            this.attr = attr;
        }

        public void setEl(String el) {
            this.el = el;
        }
    }

    @Test
    public void xmlValueIsMappedCorrectly() throws IOException {
        final RootWithJaxbAnnotations value = new RootWithJaxbAnnotations();

        final XmlMapper mapper = new XmlMapper();
        mapper.registerModule(new JaxbAnnotationModule());
        assertEquals(EXPECTED, mapper.writeValueAsString(value));
    }

    @Test
    public void jacksonXmlTextIsMappedCorrectly() throws IOException {
        final RootWithJacksonAnnotations value = new RootWithJacksonAnnotations();

        final XmlMapper mapper = new XmlMapper(new JacksonXmlModule());
        assertEquals(EXPECTED, mapper.writeValueAsString(value));
    }

}
```

If Jackson own annotations are used, it works as expected. `jacksonXmlTextIsMappedCorrectly()` succeeds. If Jaxb annotations are used (`xmlValueIsMappedCorrectly()`), Jackson creates a unwanted extra element: `text<el>text</el>` instead of `text`. If getters are removed from `JaxbChild`, it works as expected as well. However, not having getters is not a valiable solution.

**Comment by cowtowncoder** [comment]
Apologies for slow follow-up here. A way to force expected behavior is to move to accessor (`getEl()`): not sure why, but adding to field does not work (nor seems to setter, which is stranger).
I suspect this is due to odd logic JAXB annotation introspector uses to try to combine annotations, and visibility rules that are different for JAXB.

I'll see if I can find out anything more; ideally adding annotation on field would also work.

**Comment by cowtowncoder** [comment]
Ok. So the problem, fundamentally, is that `JaxbAnnotationIntrospector` gives virtual name of "value" (configurable, but this is the default) to accessor annotated with `@XmlValue`. This prevents it from using field name (`el`) to match with getter and setter. If it returned nothing, name-matching would work.

Unfortunately, returning of "value" is on purpose to make things more or less work with JSON use case and cannot be changed at this point: ideally handling would be difference depending on backend (for XML returning `null`, for others "value"), but this is not possible currently (maybe for 3.0?).

However: it is possible to already configure introspector by calling `setNameUsedForXmlValue(null)` -- this will change handling. For 2.12 I added same method in `JaxbAnnotationModule` so that it is bit more convenient:

```java
        final XmlMapper mapper = XmlMapper.builder()
                .addModule(new com.fasterxml.jackson.module.jaxb.JaxbAnnotationModule()
                        .setNameUsedForXmlValue(null)
                )
                .build(); 
```

which will give desired behavior. I added this setting to unit test as well.

Alternate for users is to just add second `@XmlValue` on other accessor(s): one on setter or getter would solve the issue.

I will mark this as resolved as I am not sure this can be further improved on short term.
